src/atomate2/abinit/files.py

```diff
@@ -139,3 +157,139 @@ def write_abinit_input_set(
         raise RuntimeError("AbinitInputSet is not valid.")
 
     ais.write_input(directory=directory, make_dir=True, overwrite=False)
+
+
+def write_mrgddb_input_set(
+    input_set_generator: MrgddbInputGenerator,
+    prev_outputs: str | Path | list[str] | None = None,
+    directory: str | Path = ".",
+) -> None:
+    """Write the mrgddb input using a given generator.
+
+    Parameters
+    ----------
+    input_set_generator
+        The input generator used to write the mrgddb inputs.
+    prev_outputs
+        The list of previous directories needed for the calculation.
+    directory
+        Directory in which to write the abinit inputs.
+    """
+    mrgis = input_set_generator.get_input_set(
+        prev_outputs=prev_outputs,
+        workdir=directory,
+    )
+    if not mrgis.validate():
+        raise RuntimeError(
+            "MrgddbInputSet is not valid. Some previous outputs \
+        do not exist."
+        )
+
+    mrgis.write_input(directory=directory, make_dir=True, overwrite=False)
+
+
+def write_anaddb_input_set(
+    structure: Structure,
+    input_set_generator: AnaddbInputGenerator,
+    prev_outputs: str | Path | list[str] | None = None,
+    directory: str | Path = ".",
+) -> None:
+    """Write the anaddb input using a given generator.
+
+    Parameters
+    ----------
+    input_set_generator
+        The input generator used to write the anaddb inputs.
+    prev_outputs
+        The list of previous directories needed for the calculation.
+    directory
+        Directory in which to write the abinit inputs.
+    """
+    anais = input_set_generator.get_input_set(
+        structure=structure,
+        prev_outputs=prev_outputs,
+    )
+    if not anais.validate():
+        raise RuntimeError(
+            "AnaddbInputSet is not valid. Some previous outputs \
+        do not exist."
+        )
+
+    anais.write_input(directory=directory, make_dir=True, overwrite=False)
+
+
+# TODO: atm does not take the directories of the generate..., run_rf,
+# store_inputs, and its own directory into account...
+@job
+def del_gzip_files(
```

Comment:
Is this used anywhere explicitly?
If not and should be added manually, could it be good to have a powerup that appends it to the Flow as a final job?

Reply:
you mean the function itself? no, I thought that the user could use this function at the end any job/flow he is running if he wants it. A powerup is a good idea to simplify its usage, I'll work on it.

Reply:
I wrote a simple powerup that returns a new Flow with this function at the end, basically what I was using when launching my own flows, is it sufficient?

Reply:
actually this only works for run_locally... the delete job is not added to the launchpad when I try it...

Reply:
> I wrote a simple powerup that returns a new Flow with this function at the end, basically what I was using when launching my own flows, is it sufficient? it won't work on Maker though...

I am not sure what you mean about this point. It should not work on Maker instances, right? You will only even need to pass the Flow that you get from the after calling the Maker.make() method.



> actually this only works for run_locally... the delete job is not added to the launchpad when I try it... EDIT: it is being added to the Launchpad if I add metadata to the whole flow after having added this delete job... I don't understand why...

This seems really weird. Since you are making a copy in the powerup, are you sure you are taking the `flow` that comes out of the powerup?